fuel/datasets/hdf5.py

```diff
@@ -564,7 +564,11 @@ def _out_of_memory_get_data(self, state=None, request=None):
             raise ValueError()
         data = []
         shapes = []
-        handle = self._file_handle
+        try:
+            handle = self._file_handle
```

Comment:
This would probably do the job, but it is somewhat inconsistent with the general idea of `@do_not_pickle_attributes` and `load`. The fact that `load` does not restore the status of the dataset is a bug, and this is probably what should be fixed.